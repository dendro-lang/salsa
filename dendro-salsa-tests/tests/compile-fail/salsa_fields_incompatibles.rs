#[salsa::jar(db = Db)]
struct Jar(InputWithBannedName1, InputWithBannedName2);

// Banned field name: `from`
#[salsa::input]
struct InputWithBannedName1 {
    from: u32,
}

// Banned field name: `new`
#[salsa::input]
struct InputWithBannedName2 {
    new: u32,
}

trait Db: salsa::DbWithJar<Jar> {}


fn main() {}