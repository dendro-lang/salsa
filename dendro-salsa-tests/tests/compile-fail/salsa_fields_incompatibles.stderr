error: the field name `from` is disallowed in salsa structs
 --> tests/compile-fail/salsa_fields_incompatibles.rs:7:5
  |
7 |     from: u32,
  |     ^^^^

error: the field name `new` is disallowed in salsa structs
  --> tests/compile-fail/salsa_fields_incompatibles.rs:13:5
   |
13 |     new: u32,
   |     ^^^

error[E0412]: cannot find type `InputWithBannedName1` in this scope
 --> tests/compile-fail/salsa_fields_incompatibles.rs:2:12
  |
2 | struct Jar(InputWithBannedName1, InputWithBannedName2);
  |            ^^^^^^^^^^^^^^^^^^^^ not found in this scope

error[E0412]: cannot find type `InputWithBannedName2` in this scope
 --> tests/compile-fail/salsa_fields_incompatibles.rs:2:34
  |
2 | struct Jar(InputWithBannedName1, InputWithBannedName2);
  |                                  ^^^^^^^^^^^^^^^^^^^^ not found in this scope
